_posts/2023-03-09-solidity-developer-survey-2022-results.md

```diff
@@ -0,0 +1,451 @@
+
+#### Other Smart Contract Languages
+
+Half of all respondents use other smart contract languages alongside Solidity. The most used other smart contract language is [Yul](https://docs.soliditylang.org/en/latest/yul.html), an intermediate language for Solidity, with 17.2%, followed by [Vyper](https://docs.vyperlang.org/en/latest/index.html), a pythonic EVM language, with 10.5%. [Cairo](https://www.cairo-lang.org/docs/) (7.1%), a STARK based language targeting StarkNet, and [Huff](https://docs.huff.sh/) (6.2%), a low level assembly language for the EVM, are also mentioned several times. Other “newcomers” like [Sway](https://fuellabs.github.io/sway/v0.35.3/book/) (2.4%) and [Fe](https://fe-lang.org/docs/index.html) (1.5%) also make it into the chart.
+
+![Other Smart Contract Languages](/img/2023/02/smart_contr_langs.png)
+
+![Solidity Developer Experience Header](/img/2023/02/header4.png)
+
+## Solidity Developer Experience
+
+76.5% of all respondents believe that the Solidity developer experience generally improved throughout the last year. 25.1% are of the opinion that they noticed a big improvement compared to the previous year.
+
+7.8% say that nothing has changed in their experience, while 0.9% of the respondents think it got worse.
+
+Compared to the previous year’s results, the share of “got worse” and “I don’t know” decreased, while “no change” slightly increased. Overall, the picture is very comparable.
+
+![Solidity Developer Experience](/img/2023/02/dev_ex.png)
+
+When getting stuck on a Solidity problem, most respondents visit Ethereum StackExchange / StackOverflow for help or search for a solution on the Internet. Many also ask their coworkers for help or watch tutorial videos. 
+
+![Solving Problems](/img/2023/02/solidity_problem.png)
+
+### Recurring Issues
+
+60% of respondents don't encounter the same or similar issues multiple times when developing in Solidity.
+
+Amongst the 40% that do, debugging issues are encountered most frequently, followed by stack-too-deep errors and bytecode size limitations.
+
+_ℹ️On the topic of debugging issues, we'd like to use the opportunity to highlight a new initiative aimed at defining a common debugging data format for languages built on top of the EVM: [ethdebug](​​https://github.com/ethdebug/format/). The end result will be a specification that will allow debuggers, analyzers, and other tools to reliably map between the EVM bytecode produced by compilers and the high-level language features. This has been a common pain point across the ecosystem for years and is becoming more pressing with the introduction of the new IR-based code generator in Solidity, which often breaks implicit assumptions tools made based on how the legacy pipeline worked. We encourage all developers working on such tools to join the working group. The group has regular bi-weekly meetings and coordinates via the [ethdebug channel](https://matrix.to/#/#ethdebug:matrix.org) on Matrix._
```

Comment:
Since most people don't know what `via-IR` is, perhaps we should say explicitly that it's the new codegen:

```suggestion
_ℹ️On the topic of debugging issues, we'd like to use the opportunity to highlight a new initiative aimed at defining a common debugging data format for languages built on top of the EVM: [ethdebug](​​https://github.com/ethdebug/format/). The end result will be a specification that will allow debuggers, analyzers, and other tools to reliably map between the EVM bytecode produced by compilers and the high-level language features. This has been a common pain point across the ecosystem for years and is becoming more pressing with the introduction of the new IR-based code generator (i.e. the `via-IR` pipeline) in Solidity, which often breaks implicit assumptions tools made based on how the legacy pipeline worked. We encourage all developers working on such tools to join the working group. The group has regular bi-weekly meetings and coordinates via the [ethdebug channel](https://matrix.to/#/#ethdebug:matrix.org) on Matrix._
```